internal/model/labels.go

```diff
@@ -59,6 +61,53 @@ func (l Labels) Clone() Labels {
 	return cp
 }
 
+// Equal returns true when the two maps of labels are equal.
+func (l Labels) Equal(labels Labels) bool {
+	for key, localV := range l {
+		v, ok := labels[key]
+		if !ok {
+			return false
+		}
+		if v.Value != localV.Value {
```

Comment:
> in cases where both are set, the Values field will be used and Value will be ignored.

The `Equal` will not follow this priority order. From the description it sounds like if `Values` is set then `Value` field doesn't matter.